tensorflow/lite/micro/kernels/fully_connected.cc

```diff
@@ -115,9 +143,18 @@ TfLiteStatus FullyConnectedEval(TfLiteContext* context, TfLiteNode* node) {
           tflite::micro::GetTensorShape(input),
           tflite::micro::GetTensorData<float>(input),
           tflite::micro::GetTensorShape(filter),
+#ifdef USE_TFLM_COMPRESSION
+          tflite::micro::GetTensorData<float>(micro_context, filter,
```

Comment:
Catching this a bit late, but in the new GetTensorData function, we check and return nullptr if tensor is nullptr. However, in the existing code, its a TFLITE_DCHECK, which means we will fail here, since we expected data. I think its better to do it that way, because otherwise we could end up trying to use the nullptr somewhere.

Reply:
This is valid for cases where the bias tensor is optional as input to the operator.

Reply:
In that case I would prefer if we have another override for the optional case. Since this will be basically changing the code even for non-compression execution (in the case globally compression is enabled), it will be safer to have the nullptr DCHECK. 

If you think that will take too long, for now we could just add a single DCHECK for the weights data in the above ifdef compression code.

Reply:
Seeing how many files use it, probably easiest to just add the override GetOptionalTensorData.

Reply:
in progress.